stubs/pyinstaller/PyInstaller/depend/analysis.pyi

```diff
@@ -0,0 +1,19 @@
+# https://pyinstaller.org/en/stable/hooks.html#the-pre-safe-import-module-psim-api-method
+
+# The documentation explicitely mentions that "Normally you do not need to know about the module-graph."
+# However, some PyiModuleGraph typed class attributes are still documented as existing in imphookapi.
+from _typeshed import Incomplete
+from typing_extensions import LiteralString
+
+class PyiModuleGraph:  # incomplete
+    def __init__(
+        self,
+        pyi_homepath: LiteralString,
```

Comment:
Here's my reasoning behind those (for anything in `imphookapi.pyi`, `analysis.pyi`, `build_main.pyi`, and `modulegraph.pyi`):
None of those are meant to be initialized by a dev using PyInstaller. Anythime you have to deal with one of those classes, it's because you were given an instance of it. The string class attributes are paths and package names found on the machine (and not something using user input), so I believe they are "safe literals".

However, some of those attributes are actually directly taken from __init__ parameters, so I made those LiteralStrings as well to stay consistent (I shouldn't assign a `str` to a `LiteralString`. As I mentioned, the dev has no reason to initialize those classes (ie: `PostGraphAPI(...)`), so it doesn't matter much if it's too restrictive. In fact, the only reason I'm defining the __init__ method is so that stubtest doesn't whine about missing parameters, or pyright about missing parameter types.

If I had a way to not expose the __init__ of those classes in the first place, I'd prefer that.

If you think we should just simplify it all to use `str`, I can understand that as well, I don't have an example of a use-case where those class attributes *need* to be literal.

Reply:
I think I'd prefer it if we went with `str` for now, if that's okay. `LiteralString` is a very new type, and I don't feel like I have a good idea about how using it this widely might affect usability of the package :)